src/source/src/row_id.rs

```diff
@@ -68,14 +68,20 @@ impl RowIdGenerator {
             self.last_duration_ms = current_duration;
             self.sequence = 0;
         }
-        assert!(self.sequence < SEQUENCE_UPPER_BOUND);
 
-        let row_id = self.last_duration_ms << TIMESTAMP_SHIFT_BITS
-            | (self.worker_id << WORKER_ID_SHIFT_BITS) as i64
-            | self.sequence as i64;
-        self.sequence += 1;
+        if self.sequence < SEQUENCE_UPPER_BOUND {
+            let row_id = self.last_duration_ms << TIMESTAMP_SHIFT_BITS
+                | (self.worker_id << WORKER_ID_SHIFT_BITS) as i64
+                | self.sequence as i64;
+            self.sequence += 1;
 
-        row_id
+            row_id
+        } else {
+            // If the sequence reaches the upper bound, spin loop here and wait for next
+            // millisecond.
+            std::thread::sleep(std::time::Duration::from_millis(1));
```

Comment:
1ms seems too long? 🤣

Reply:
We are using ms as timestamp in row-id. Seems okay?

Reply:
For example when the sequence runs out at the tick of 1.9ms, then we only need to wait for 100us for next sequence number.

Reply:
Changed to sleep exact time interval. 😊